doc/spec.md

```diff
@@ -3691,15 +3713,45 @@ x                                        # set([1, 3])
 <a id='set·intersection'></a>
 ### set·intersection
 
-`S.intersection(y)` returns a new set into which have been inserted all the elements of set S which are also in y.
+`S.intersection(*others)` returns a new set containing those elements that the
+set `S` and all of the iterable sequences `*others` have in common.
 
-y can be any type of iterable (e.g. set, list, tuple).
+It is permissible to call `intersection` without any arguments; this returns a
+copy of the set `S`.
+
+`intersection` fails if any element of any of the `*others` is unhashable.
 
 ```python
-x = set([1, 2, 3])
-x.intersection([3, 4, 5])                # set([3])
+set([1, 2]).intersection([2, 3])             # set([2])
+set([1, 2, 3]).intersection([0, 1], [1, 2])  # set([1])
+```
+
+<a id='set·intersection_update'></a>
+### set·intersection\_update
+
+`S.intersection_update(*others)` removes from the set `S` any elements not found
+in at least one of the iterable sequences `*others`. It returns `None`.
+
+It is permissible to call `intersection_update` without any arguments; this
+leaves the set `S` unchanged.
+
+`intersection_update` fails if the set `S` is frozen or has active iterators, or
+if any element of any of the `*others` is unhashable.
```

Comment:
For the record, this precludes a short cut when S is empty.

Reply:
That seems reasonable? We usually prohibit no-op mutation attempts - for example, if a set is immutable, we don't allow trying to update it with an element that it already contains.